docs(releases): stellar GMP v1.1.0 and ITS v1.1.0 (#559)
axelarnetwork/axelar-contract-deployments: releases/stellar/2025-02-GMP-v1.1.0.md

# Stellar GMP v1.1.0

|                | **Owner**                            |
| -------------- | ------------------------------------ |
| **Created By** | @nbayindirli (<noah@interoplabs.io>)       |
| **Deployment** | @RiceAndMeet (<steven@interoplabs.io>) |

| **Network**          | **Deployment Status** | **Date** |
| -------------------- | --------------------- | -------- |
| **Devnet Amplifier** | -                     | TBD      |
| **Stagenet**         | -                     | TBD      |
| **Testnet**          | -                     | TBD      |
| **Mainnet**          | -                     | TBD      |

- [v1.1.0 Crates](https://crates.io/crates/stellar-axelar-gateway/1.1.0)

> - [v1.1.0 Releases](https://github.com/axelarnetwork/axelar-cgp-stellar/releases/tag/stellar-axelar-gateway-v1.1.0)

Reviewer: Note, the repo name will be changed to `axelar-amplifier-stellar`: https://axelarnetwork.atlassian.net/browse/AXE-7986

    --version 1.1.0
```

3. Retrieve full list of operators used by the relayer.

| Network              | `axelar_operators storage` |
| -------------------- | ---------------------------------------------------------- |
| **Devnet-amplifier** | [source](https://stellar.expert/explorer/testnet/contract/CABRZHKAJVJFX5IZWL7KPA6ZEX6I347W7GBXAUX6WVOGUUBRWCDGOUSI/storage) |
| **Stagenet**         | `TBD`                                                      |
| **Testnet**          | `TBD`                                                      |
| **Mainnet**          | `TBD`                                                      |

| Network              | `operator addresses`|
| -------------------- | ---------------------------------------------------------- |
| **Devnet-amplifier** | `GBM47ULSN4OEQNPECDIMZSDMK6AH2RNICV2VGOYPBVJ4LBFCFFQP2BXM` |
| **Stagenet**         | `TBD`                                                      |
| **Testnet**          | `TBD`                                                      |
| **Mainnet**          | `TBD`                                                      |

4. Upgrade `axelar_operators` and migrate storage schema

> | Network              | `deployer address`                                         |

Reviewer: why `deployer address` table is here - is this `stellar_deployer_key`?

## Background

- This is the v1.1.0 Stellar GMP release.
- This release contains no breaking changes.

Contract changes in the release:

1. Add `contractstorage` attribute macro, enabling named keys, typed values, storage method specification in the storage layout, storage schema validation, and built-in TTL extension. Use this macro in lieu of Soroban's contracttype macro on storage enums and calling storage on the env.
2. Storage TTL will now be extended automatically upon storage access due to this macro.
3. Migrate the `axelar-operators` contract's storage schema key from '`Operators`' to '`Operator`'.
4. Make the `gas_token` parameter for ITS calls optional.

## Deployment

Create an `.env` config. `CHAIN` should be set to `stellar` for `mainnet`, and `stellar-2024-q4` for all other networks.

```yaml
# Change `PRIVATE_KEY in `.env` to Stellar
PRIVATE_KEY=<stellar_deployer_key>
ENV=<devnet-amplifier|stagenet|testnet|mainnet>
CHAIN=<stellar-2024-q4|stellar>
```

A Stellar chain config should already exist under a `${ENV}.json` file under `chains` key.

### Steps

1. Request Tokens from Faucet

```bash
node stellar/faucet.js
```

2. Upgrade the `Upgrader` contract

```bash

> node stellar/deploy-contract.js upgrade Upgrader \
>     --version v1.1.0

Reviewer: ```suggestion
node stellar/deploy-contract.js upgrade Upgrader --version v1.1.0
```

| **Testnet**          | `TBD`                                                      |
| **Mainnet**          | `TBD`                                                      |

5. Upgrade `AxelarOperators` and migrate storage schema

| Network              | `deployer address`                                         |
| -------------------- | ---------------------------------------------------------- |
| **Devnet-amplifier** | `GDUITDF2LI3R5HM4KYRLLNRLEWKYBFVZVOEB6HSL7EOW2KO2LD6V4GPM` |
| **Stagenet**         | `TBD`                                                      |
| **Testnet**          | `TBD`                                                      |
| **Mainnet**          | `TBD`                                                      |

```bash
node stellar/deploy-contract.js upgrade AxelarOperators --version v1.1.0 --migration-data '["<operatorN>"]'
```

6. Revisit the AxelarOperators storage via the explorer link above.

- The value `Operators` should now read `Operator`.
- Its key(s) should be set the the operator address retrieved above.

7. Upgrade the `AxelarGasService` contract

```bash
node stellar/deploy-contract.js upgrade AxelarGasService --version v1.1.0
```

> 

Reviewer: AxelarExample upgrade with dummy ITS is missing here - we have 2 steps for the AxelarExample, 1) deploy it with dummy ITS address, 2) deploy it again with ITS address. AxelarExample is being used for `stellar/gmp.js` command

> # Stellar GMP v1.1.0

Reviewer: file name to `2025-03-`

| -------------------- | --------------------- | -------- |
| **Devnet Amplifier** | -                     | TBD      |
| **Stagenet**         | -                     | TBD      |
| **Testnet**          | -                     | TBD      |
| **Mainnet**          | -                     | TBD      |

- [v1.1.0 Crates](https://crates.io/crates/stellar-axelar-gateway/1.1.0)
- [v1.1.0 Releases](https://github.com/axelarnetwork/axelar-amplifier-stellar/releases/tag/stellar-axelar-gateway-v1.1.0)

## Background

- This is the v1.1.0 Stellar GMP release.
- This release contains no breaking changes.

Contract changes in the release:

1. Add `contractstorage` attribute macro, enabling named keys, typed values, storage method specification in the storage layout, storage schema validation, and built-in TTL extension. Use this macro in lieu of Soroban's contracttype macro on storage enums and calling storage on the env.
2. Storage TTL will now be extended automatically upon storage access due to this macro.
3. Migrate the `axelar-operators` contract's storage schema key from '`Operators`' to '`Operator`'.
4. Make the `gas_token` parameter for ITS calls optional.

## Deployment

> Create an `.env` config. `CHAIN` should be set to `stellar` for `mainnet`, and `stellar-2024-q4` for all other networks.

Reviewer: ```suggestion
Create an `.env` config. `CHAIN` should be set to `stellar` for `mainnet`, and `stellar-2025-q1` for all other networks.
```

```yaml
# Change `PRIVATE_KEY in `.env` to Stellar
PRIVATE_KEY=<stellar_deployer_key>
ENV=<devnet-amplifier|stagenet|testnet|mainnet>

> CHAIN=<stellar-2024-q4|stellar>

Reviewer: ```suggestion
CHAIN=<stellar-2025-q1|stellar>
```

| **Mainnet**          | `TBD`                                                      | // TODO: ADD

5. Upgrade the `AxelarGateway` contract & migrate storage schema

```bash
node stellar/deploy-contract.js upgrade AxelarGateway --version v1.1.0 --migration-data '[("<sourceChainN>", "<messageIdN>")]'
```

6. Retrieve full list of operators used by the relayer

| Network              | `AxelarOperators storage`                                  |
| -------------------- | ---------------------------------------------------------- |
| **Devnet-amplifier** | [source](https://stellar.expert/explorer/testnet/contract/CABRZHKAJVJFX5IZWL7KPA6ZEX6I347W7GBXAUX6WVOGUUBRWCDGOUSI/storage)                   |
| **Stagenet**         | `TBD`                                                      | // TODO: ADD
| **Testnet**          | `TBD`                                                      | // TODO: ADD
| **Mainnet**          | `TBD`                                                      | // TODO: ADD

| Network              | `AxelarOperators migration data`                                 |
| -------------------- | ---------------------------------------------------------------- |

> | **Devnet-amplifier** | `'["GBM47ULSN4OEQNPECDIMZSDMK6AH2RNICV2VGOYPBVJ4LBFCFFQP2BXM"]'` |

Reviewer: `TBD` as well for now?

# Stellar GMP v1.1.0

|                | **Owner**                            |
| -------------- | ------------------------------------ |
| **Created By** | @nbayindirli (<noah@interoplabs.io>)       |
| **Deployment** | @RiceAndMeet (<steven@interoplabs.io>) |

| **Network**          | **Deployment Status** | **Date** |
| -------------------- | --------------------- | -------- |
| **Devnet Amplifier** | -                     | TBD      |
| **Stagenet**         | -                     | TBD      |
| **Testnet**          | -                     | TBD      |
| **Mainnet**          | -                     | TBD      |

- [v1.1.0 Crates](https://crates.io/crates/stellar-axelar-gateway/1.1.0)
- [v1.1.0 Releases](https://github.com/axelarnetwork/axelar-cgp-stellar/releases/tag/stellar-axelar-gateway-v1.1.0)

## Background

- This is the v1.1.0 Stellar GMP release.

Contract changes in the release:

> 1. Add contractstorage attribute macro, enabling named keys, typed values, storage method specification in the storage layout, storage schema validation, and built-in TTL extension. Use this macro in lieu of Soroban's contracttype macro on storage enums and calling storage on the env.

Reviewer 2: Mention that there are no breaking changes, and highlight that TTL will be extended automatically whenever storage is accessed, which is a public facing change

> 2. Add only_owner and only_operator macros, reducing boilerplate for auth-checks and increasing code readability. Use this macro in lieu of directly checking the owner or operator.

Reviewer 2: Not relevant to external users

3. Migrate the axelar-operators contract's storage schema key from 'Operators' to 'Operator'.

> 4. Add 'stellar/operators.js is-operator', 'stellar/operators.js add-operator', 'stellar/operators.js remove-operator', 'stellar/operators.js collect-fees', 'stellar/operators.js refund', and 'stellar/operators.js execute' commands to interact with the axelar-operators contract.

Reviewer 2: Minor changes not worth mentioning. Only document breaking changes, migrations, or new features in the contracts. The scripts usage will be showcased below

> Deployment changes in the release:

Reviewer 2: Same, not needed

1. Add 'stellar/gateway.js execute' command to execute GMP messages from the CLI.
2. Make gas-token-address and gas-fee-amount optional for 'stellar/gmp.js send' command.
3. Add 'stellar/contract.js paused', 'stellar/contract.js pause', 'stellar/contract.js unpause' commands to check the pausable state of a Stellar contract.
4. Make minimum-rotation-delay optional for 'stellar/deploy-contract.js' commands.
5. Add 'evm/gateway.js --action isContractCallApproved' command to verify if a gateway contract call is approved from the CLI.
6. Add 'evm/gateway.js --action isMessageApproved' command to verify if a gateway message is approved from the CLI.

## Deployment

> Create an `.env` config. `CHAIN` should be set to `stellar` for mainnet, and `stellar-2025-q1` for all other networks.

Reviewer 2: ```suggestion
Create an `.env` config. `CHAIN` should be set to `stellar` for mainnet, and `stellar-2024-q4` for all other networks.
```
Same chain name must be used as the existing deployment being upgraded. 2024 Q4 because that's when the Stellar testnet was launched

```yaml
PRIVATE_KEY=xyz
ENV=xyz
CHAIN=xyz
```

An initial Stellar chain config needs to be added to `${ENV}.json` file under `chains` key.

### Devnet Amplifier / Stagenet / Testnet

```bash

> \"$CHAIN\": {

Reviewer 2: This is not a new deployment of Stellar, so this can be assumed to exist in the configs

    "name": "Stellar",
    "axelarId": \"$CHAIN\",
    "rpc": "https://soroban-testnet.stellar.org",
    "horizonRpc": "https://horizon-testnet.stellar.org",
    "networkType": "testnet",
    "chainType": "stellar",
    "decimals": 7,
    "finality": "1",
    "approxFinalityWaitTime": 1,
    "tokenSymbol": "XLM",
    "tokenAddress": "CDLZFC3SYJYDZT7K67VZ75HPJVIEUVNIXF47ZG2FB2RMQQVU2HHGCYSC",
    "explorer": {
      "name": "Stellar Expert",
      "url": "https://stellar.expert/explorer/testnet"
    },
  "contracts": {}
  }
```

### Mainnet

```bash

> "stellar": {

Reviewer 2: Same

    "name": "Stellar",
    "axelarId": "stellar",
    "rpc": "https://mainnet.sorobanrpc.com",
    "horizonRpc": "https://horizon.stellar.org",
    "networkType": "mainnet",
    "chainType": "stellar",
    "decimals": 7,
    "finality": "1",
    "approxFinalityWaitTime": 1,
    "tokenSymbol": "XLM",
    "tokenAddress": "CAS3J7GYLGXMF6TDJBBYYSE3HQ6BBSMLNUQ34T6TZMYMW2EVH34XOWMA",
    "explorer": {
      "name": "Stellar Expert",
      "url": "https://stellar.expert/explorer/public"
    },
  "contracts": {}
  }
```

1. Request Tokens from Faucet

```bash
node stellar/faucet.js
```

2. Retrieve full list of operators used by the relayer.

| Network              | `axelar_operators address`|
| -------------------- | ---------------------------------------------------------- |

> | **Devnet-amplifier** | `CABRZHKAJVJFX5IZWL7KPA6ZEX6I347W7GBXAUX6WVOGUUBRWCDGOUSI` |

Reviewer 2: This is stored in the config, no need to mention here. Only inputs that need to be provided are necessary

| **Stagenet**         | `TBD`                                                      |
| **Testnet**          | `TBD`                                                      |
| **Mainnet**          | `TBD`                                                      |

| Network              | `axelar_operators storage` |
| -------------------- | ---------------------------------------------------------- |

> | **Devnet-amplifier** | [source](https://stellar.expert/explorer/testnet/contract/CABRZHKAJVJFX5IZWL7KPA6ZEX6I347W7GBXAUX6WVOGUUBRWCDGOUSI/storage) |

Reviewer 2: Neat. Can you find out if they actually store the keys in storage as well, or is it just the explorer indexing it. Wondering since soroban sdk doesn't allow iterating over the keys

| **Stagenet**         | `TBD`                                                      |
| **Testnet**          | `TBD`                                                      |
| **Mainnet**          | `TBD`                                                      |

| Network              | `OPERATOR_ADDRESS`|
| -------------------- | ---------------------------------------------------------- |
| **Devnet-amplifier** | `GBM47ULSN4OEQNPECDIMZSDMK6AH2RNICV2VGOYPBVJ4LBFCFFQP2BXM` |
| **Stagenet**         | `TBD`                                                      |
| **Testnet**          | `TBD`                                                      |
| **Mainnet**          | `TBD`                                                      |

3. Upgrade axelar_operators and migrate storage schema

> | Network              | `deployer address`                                         |

Reviewer 2: also not needed

| -------------------- | ---------------------------------------------------------- |
| **Devnet-amplifier** | `GDUITDF2LI3R5HM4KYRLLNRLEWKYBFVZVOEB6HSL7EOW2KO2LD6V4GPM` |
| **Stagenet**         | `TBD`                                                      |
| **Testnet**          | `TBD`                                                      |
| **Mainnet**          | `TBD`                                                      |

```bash
node stellar/deploy-contract.js upgrade axelar_operators \

>     --new-version 'v1.1.0' \

Reviewer 2: ```suggestion
    --new-version v1.1.0 \
```
maybe

>     --migration-data OPERATOR_ADDRESS_A,OPERATOR_ADDRESS_B,OPERATOR_ADDRESS_C

Reviewer 2: is this correct encoding for the migration data? Did you try upgrading?
```suggestion
    --migration-data [operator1,operator2,operator3]
```

```

## Checklist

The following checks should be performed after the rollout:

### Axelar Operators Calls

1. Verify the axelar_operators storage has migrated as expected.

> | Network              | `OPERATOR_ADDRESS`|

Reviewer 2: same, not needed, already declared

| -------------------- | ---------------------------------------------------------- |
| **Devnet-amplifier** | `GBM47ULSN4OEQNPECDIMZSDMK6AH2RNICV2VGOYPBVJ4LBFCFFQP2BXM` |
| **Stagenet**         | `TBD`                                                      |
| **Testnet**          | `TBD`                                                      |
| **Mainnet**          | `TBD`                                                      |

Run the following command for each OPERATOR_ADDRESS:
```bash

> node stellar/operators.js is-operator OPERATOR_ADDRESS_A
> node stellar/operators.js is-operator OPERATOR_ADDRESS_B
> node stellar/operators.js is-operator OPERATOR_ADDRESS_C

Reviewer 2: ```suggestion
node stellar/operators.js is-operator OPERATOR_ADDRESS_$i
```
